Python/ast.c

```
@@ -903,25 +907,37 @@ static const char * const FORBIDDEN[] = {
     "None",
     "True",
     "False",
+    "__debug__",
     NULL,
 };
 
+static int
+is_forbidden_name(const char *name)
```

Can we not duplicate this by having ast_for_call turn the name into an identifier and passing it to `forbidden_name`?

---

I tried this initially. But this needed to add more complex code at calling places if we want to raise the same error `"expression cannot contain assignment"` for both `f(0 = a)` and `f(False = a)`.

But if `"cannot assign to False"` is good enough or even better for the latter case, the code can be simplified.